File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/UpdateBigSegmentContextTargetsExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class UpdateBigSegmentContextTargetsExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var included = new SegmentUserList();
        included.add(List.of ());
        included.remove(List.of ());

        var excluded = new SegmentUserList();
        excluded.add(List.of ());
        excluded.remove(List.of ());

        var segmentUserState = new SegmentUserState();
        segmentUserState.included(included);
        segmentUserState.excluded(excluded);

        try
        {
            new SegmentsApi(config).updateBigSegmentContextTargets(
                "projectKey_string", // projectKey
                "environmentKey_string", // environmentKey
                "segmentKey_string", // segmentKey
                segmentUserState
            );
        } catch (ApiException e) {
            System.err.println("Exception when calling SegmentsApi#updateBigSegmentContextTargets");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/UpdateBigSegmentContextTargetsExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class UpdateBigSegmentContextTargetsExample
{
    fun updateBigSegmentContextTargets()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val included = SegmentUserList(
            add = listOf (),
            remove = listOf (),
        )

        val excluded = SegmentUserList(
            add = listOf (),
            remove = listOf (),
        )

        val segmentUserState = SegmentUserState(
            included = included,
            excluded = excluded,
        )

        try
        {
            SegmentsApi().updateBigSegmentContextTargets(
                projectKey = null,
                environmentKey = null,
                segmentKey = null,
                segmentUserState = segmentUserState,
            )
        } catch (e: ClientException) {
            println("4xx response calling SegmentsApi#updateBigSegmentContextTargets")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling SegmentsApi#updateBigSegmentContextTargets")
            e.printStackTrace()
        }
    }
}
